src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,88 @@
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import ReactNode = React.ReactNode
+import { UIComponent, customPropTypes, IRenderResultConfig } from '../../lib'
+import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
+import { Extendable } from '../../../types/utils'
+import List, { ListItem } from '../List'
+import Popup from '../Popup'
+export interface IContextMenuProps {
+  as?: any
+  className?: string
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A contextualMenu.
+ * @accessibility This is example usage of the accessibility tag.
+ * This should be replaced with the actual description after the PR is merged
+ */
+class ContextMenu extends UIComponent<Extendable<IContextMenuProps>, any> {
+  static displayName = 'ContextMenu'
+
+  static className = 'ui-contextmenu'
+
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** The tree of menu containing submenus and it's submenus and so on. */
+    items: PropTypes.arrayOf(PropTypes.object),
+
+    /** Function passed which needs to be invoked when menuitem has no submenu */
```

Comment:
Don't really understand comment.  For menu, we have onclick on the menu-items.  Can't we do same here?  It's very likely that different menu items what to call different functions....  (and nit - end comment with '.')

Reply:
Yes, but this just handles the case of one callback function for all menu-items. The `onClick` on all items will also work at the same time.